runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingModeExecutionContext.java

```diff
@@ -292,41 +213,54 @@ protected SideInputReader getSideInputReaderForViews(
    * Fetches the requested sideInput, and maintains a view of the cache that doesn't remove items
    * until the active work item is finished.
    *
-   * <p>If the side input was not ready, throws {@code IllegalStateException} if the state is
-   * {@literal CACHED_IN_WORKITEM} or returns null otherwise.
-   *
-   * <p>If the side input was ready and null, returns {@literal Optional.absent()}. If the side
-   * input was ready and non-null returns {@literal Optional.present(...)}.
+   * <p>If the side input cached, throws {@code IllegalStateException} if the state is {@literal
+   * CACHED_IN_WORK_ITEM} or returns {@link SideInput<T>} which contains {@link Optional<T>}.
    */
-  private @Nullable <T> Optional<T> fetchSideInput(
+  @SuppressWarnings({"deprecation", "unchecked"})
```

Comment:
can you move the unchecked suppression to where cast is made (similar to before)?

what is deprecation suppression required for?

Reply:
`getTagInternal` is deprecated with message:
```
this method will be removed entirely. The {@link PCollection} underlying a side
input, is part of the side input's specification with a {@link ParDo} transform, which will
obtain that information via a package-private channel.
```

done.